antenna/breakpad_resource.py

```diff
@@ -288,51 +288,55 @@ def get_throttle_result(self, raw_crash):
         :returns tuple: ``(result, rule_name, percentage)``
 
         """
-        # If the raw_crash has a uuid, then that implies throttling, so return
-        # that.
-        if 'uuid' in raw_crash:
-            crash_id = raw_crash['uuid']
-            if crash_id[-7] in (str(ACCEPT), str(DEFER)):
-                result = int(crash_id[-7])
+        # FIXME(willkg): We should move some of these things into throttle
+        # rules instead of having them here.
+
+        # Throttle the crash for an initial result
+        result, rule_name, throttle_rate = self.throttler.throttle(raw_crash)
+
+        # If we didn't reject the crash, then go through it and override the
+        # throttle result in certain circumstances.
+        if result != REJECT:
+            if 'uuid' in raw_crash:
+                # If the raw_crash has a uuid, then that implies throttling, so return
+                # that
+                crash_id = raw_crash['uuid']
+                if crash_id[-7] in (str(ACCEPT), str(DEFER)):
+                    result = int(crash_id[-7])
+                    rule_name = 'FROM_CRASHID'
+                    throttle_rate = 100
+
+            elif 'legacy_processing' in raw_crash and 'throttle_rate' in raw_crash:
+                # If we have throttle results for this crash and the values are
+                # valid, then use those instead of the throttle result
+                try:
+                    temp_result = int(raw_crash['legacy_processing'])
+                    if temp_result not in (ACCEPT, DEFER):
+                        raise ValueError('Result is not a valid value: %r', result)
+
+                    temp_throttle_rate = int(raw_crash['throttle_rate'])
+                    if not (0 <= temp_throttle_rate <= 100):
+                        raise ValueError('Throttle rate is not a valid value: %r', result)
+
+                    # The two values are valid, so this becomes our new
+                    # throttle result
+                    result = temp_result
+                    rule_name = 'ALREADY_THROTTLED'
+                    throttle_rate = temp_throttle_rate
+
+                except ValueError:
+                    # If we've gotten a ValueError, it means one or both of the
+                    # values is bad and we should ignore it and move forward
+                    mymetrics.incr('throttle.bad_throttle_values')
+
+            elif raw_crash.get('Throttleable', None) == '0':
+                # If the raw crash has ``Throttleable=0``, then it was
+                # submitted manually, so accept the crash instead of the
+                # throttle result
+                result = ACCEPT
+                rule_name = 'THROTTLEABLE_0'
```

Comment:
This block of code diffs poorly because it's all indented one level under a new `if not REJECT`. Sorry. :(